utils/unittest/unittest.go

```diff
@@ -95,6 +95,15 @@ func RequireCloseBefore(t testing.TB, c <-chan struct{}, duration time.Duration,
 	}
 }
 
+// RequireClosed is a test helper function that fails the test if channel `ch` is not closed.
+func RequireClosed(t *testing.T, ch <-chan struct{}, message string) {
+	select {
+	case <-ch:
+	default:
+		require.Fail(t, "channel is not closed: "+message)
+	}
+}
```

Comment:
Note: this is *guaranteed* to select from the channel if it is closed, and will *only* run the default case if nothing is available to be read from the channel: https://stackoverflow.com/questions/45580151/priority-of-case-versus-default-in-golang-select-statements